python/graphstorm/gpartition/dist_partition_graph.py

```diff
@@ -0,0 +1,154 @@
+
+from threading import Thread
+
+from .random_partition import LocalRandomPartitioner
+from ..utils import get_log_level
+
+def run_build_dglgraph(
+        input_data_path,
+        partitions_dir,
+        ip_list,
+        output_path,
+        metadata_filename,
+        dgl_tool_path):
+        """ Build DistDGL Graph
+
+        Parameters
+        ----------
+        input_data_path: str
+            Path to the input graph data
+        partitions_dir: str
+            Path to Per-node-type partid mappings.
+        ip_list: str
+            Path to a file storing ips of instances
+        output_path: str
+            Output Path
+        metadata_filename: str
+            The filename for the graph partitioning metadata file we'll use to determine data sources.
+        """
+        # Get the python interpreter used right now.
+        # If we can not get it we go with the default `python3`
+        python_bin = sys.executable \
+            if sys.executable is not None and sys.executable != "" \
+            else "python3 "
+        state_q = queue.Queue()
+
+        launch_cmd = ["python3", f"{dgl_tool_path}/dispatch_data.py",
+            "--in-dir", input_data_path,
+            "--metadata-filename", metadata_filename,
+            "--partitions-dir", partitions_dir,
+            "--ip-config", ip_list,
+            "--out-dir", output_path,
+            "--ssh-port", "22",
+            "--python-path", f"{python_bin}",
+            "--log-level", logging.getLevelName(logging.root.getEffectiveLevel()),
+            "--save-orig-nids",
+            "--save-orig-eids"]
+
+        # thread func to run the job
+        def run(cmd, state_q):
+            try:
+                subprocess.check_call(cmd, shell=False)
+                state_q.put(0)
+            except subprocess.CalledProcessError as err:
+                logging.error("Called process error %s", err)
+                state_q.put(err.returncode)
+            except Exception: # pylint: disable=broad-exception-caught
+                state_q.put(-1)
+
+        # launch postprocess task
+        thread = Thread(target=run, args=(launch_cmd, state_q,), daemon=True)
+        thread.start()
+        # sleep for a while in case of ssh is rejected by peer due to busy connection
+        time.sleep(0.2)
+
+        thread.join()
+        err_code = state_q.get()
+        if err_code != 0:
+            raise RuntimeError("build dglgrah failed")
+
+
+def main(args):
+    logging.basicConfig(level=get_log_level(args.logging_level))
+    output_path = args.output_path
+    metadata_file=args.meta_info_path
+    local_output_path = os.path.join(output_path, "tmp")
+
+    part_start = time.time()
+    if args.part_algorithm == "random":
+        partitioner = LocalRandomPartitioner(metadata_file, local_output_path)
+        local_partition_path = partitioner.create_partitions(args.num_parts)
+    else:
+        raise RuntimeError(f"Unknow partition algorithm {args.part_algorighm}")
```

Comment:
See comment in `random_partition.py`, let's write this more as 

```
if args.part_algorithm == "random":
    partitioner = RandomPartitionAlgorithm(metadata_file, local_output_path)
elif args.part_algorithm == "range":
    partitioner = RangePartitionAlgorithm(metadata_file, local_output_path)
else:
    raise NotImplementedError()

local_partition_path = partitioner.create_partitions(args.num_parts)
```

We should also have a contract that `partitioner.create_partitions()` is the main common interface between different partition algorithm implementations, and what the expected output is from calling that function, that is, one partition assignment file named this specific way is created in this specific path (and write it in docstring)